server/sonar-server/src/main/java/org/sonar/server/measure/ws/SearchAction.java

```diff
@@ -225,10 +251,39 @@ private SearchWsResponse buildResponse() {
           SearchWsResponse::newBuilder,
           SearchWsResponse.Builder::addComponents,
           (result1, result2) -> {
-            throw new IllegalStateException("Parallel execution forbidden while build SearchWsResponse");
+            throw new IllegalStateException("Parallel execution forbidden while building SearchWsResponse");
           },
           SearchWsResponse.Builder::build));
     }
 
+    private List<MeasureDto> buildBestMeasures() {
+      Set<MetricDto> metricsWithBestValue = metrics.stream()
+        .filter(metric -> metric.isOptimizedBestValue() && metric.getBestValue() != null)
+        .collect(Collectors.toSet());
+
+      Multimap<String, WsMeasures.Period> wsPeriodsByProjectUuid = snapshots.stream().collect(Collector.of(
```

Comment:
as discussed offline, it's a complexity that is very valuable for the WS consumer